samples/nrf9160/serial_lte_modem/src/lwm2m/th_lwm2m_service.c

```diff
@@ -0,0 +1,727 @@
+};
+
+static const char client_psk_id[] = "Client_identity";
+#endif /* CONFIG_LWM2M_DTLS_SUPPORT */
+
+#if defined(CONFIG_LWM2M_FIRMWARE_UPDATE_OBJ_SUPPORT)
+static u8_t firmware_buf[64];
+#endif
+
+/* TODO: Move to a pre write hook that can handle ret codes once available */
+static int light_control_write_cb(u16_t obj_inst_id, u16_t res_id,
+	u16_t res_inst_id,
+	u8_t *data, u16_t data_len,
+	bool last_block, size_t total_size)
+{
+	u8_t buf[5];
+
+	LOG_INF("Light Control: write %d", obj_inst_id);
+
+	buf[0] = (IPSO_OBJECT_LIGHT_CONTROL_ID >> 8) & 0xFF;
+	buf[1] = (IPSO_OBJECT_LIGHT_CONTROL_ID >> 0) & 0xFF;
+	buf[2] = (obj_inst_id >> 8) & 0xFF;
+	buf[3] = (obj_inst_id >> 0) & 0xFF;
+	buf[4] = *(u8_t *) data;
+	(void)inter_connect_notify(NOT_TYPE_LWM2M_OBJECT, buf, 5);
+	/* TODO: Move to be set by an internal post write function */
+	if (obj_inst_id == 0)
+		lwm2m_engine_set_s32("3311/0/5852", 0);
+	if (obj_inst_id == 1)
+		lwm2m_engine_set_s32("3311/1/5852", 0);
+	return 0;
+}
+
+static int light_control_create_cb(u16_t obj_inst_id)
+{
+	LOG_INF("Light Control: create %d", obj_inst_id);
+	return 0;
+}
+
+static int light_control_delete_cb(u16_t obj_inst_id)
+{
+	LOG_INF("Light Control: delete %d", obj_inst_id);
+	return 0;
+}
+
+static int device_reboot_cb(u16_t obj_inst_id)
+{
+	LOG_INF("DEVICE: REBOOT");
+	/* Add an error for testing */
+	lwm2m_device_add_err(LWM2M_DEVICE_ERROR_LOW_POWER);
+	/* Change the battery voltage for testing */
+	lwm2m_engine_set_s32("3/0/7/0", (bat_mv - 1));
+
+	return 0;
+}
+
+static int device_factory_default_cb(u16_t obj_inst_id)
+{
+	LOG_INF("DEVICE: FACTORY DEFAULT");
+	/* Add an error for testing */
+	lwm2m_device_add_err(LWM2M_DEVICE_ERROR_GPS_FAILURE);
+	/* Change the USB current for testing */
+	lwm2m_engine_set_s32("3/0/8/1", (usb_ma - 1));
+	return 0;
+}
+
+#if defined(CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_SUPPORT)
+static int firmware_update_cb(u16_t obj_inst_id)
+{
+	LOG_DBG("UPDATE");
+
+	/* TODO: kick off update process */
+
+	/* If success, set the update result as RESULT_SUCCESS.
+	 * In reality, it should be set at function lwm2m_setup()
+	 */
+	lwm2m_engine_set_u8("5/0/3", STATE_IDLE);
+	lwm2m_engine_set_u8("5/0/5", RESULT_SUCCESS);
+	return 0;
+}
+#endif
+
+#if defined(CONFIG_LWM2M_FIRMWARE_UPDATE_OBJ_SUPPORT)
+static void *firmware_get_buf(u16_t obj_inst_id, size_t *data_len)
+{
+	*data_len = sizeof(firmware_buf);
+	return firmware_buf;
+}
+
+static int firmware_block_received_cb(u16_t obj_inst_id,
+	u8_t *data, u16_t data_len,
+	bool last_block, size_t total_size)
+{
+	LOG_INF("FIRMWARE: BLOCK RECEIVED: len:%u last_block:%d",
+		data_len, last_block);
+	return 0;
+}
+#endif
+
+static int lwm2m_setup(void)
+{
+	int ret;
+	char *server_url;
+	u16_t server_url_len;
+	u8_t server_url_flags;
+	static bool object_instances_created;
+
+	/* setup SECURITY object */
+
+	/* Server URL */
+	ret = lwm2m_engine_get_res_data("0/0/0",
+		(void **)&server_url, &server_url_len,
```

Comment:
Alignment. In general, we tend to align with the opening bracket when arguments do not fit into a single line, no need to repeat this everywhere, just please fix wherever you spot it.

Reply:
Updated